.github/workflows/PR_clang-format_test.yml

```diff
@@ -0,0 +1,171 @@
+        script: |
+            github.issues.addLabels({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              labels: ['AT: WIP']
+            })
+
+#   Checkout this users SST-core repo/branch
+    - name: Checkout Users SST-Core source
+      uses: actions/checkout@v2
+      with:
+        path: ./sst-core_source
+
+#   Running the github action for clang format
+    - name: Run Action clang-format-lint
+      uses: DoozyX/clang-format-lint-action@v0.12
+      with:
+        source: './sst-core_source/'
+        exclude: './sst-core_source/src/sst/core/libltdl ./sst-core_source/external ./sst-core_source/build'
+        extensions: 'h,cc'
+        clangFormatVersion: 12
+        inplace: False
+
+#############################################
+# LABEL RESULTS - USING actions/github-script
+#############################################
+
+# Label the Results - FOR FAILURE
+    - name: ADD FAIL + WIP Label Results for Failure
+      uses: actions/github-script@v4
+      with:
+        github-token: ${{secrets.GITHUB_TOKEN}}
+        script: |
+            github.issues.addLabels({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              labels: ['AT: CLANG-FORMAT FAIL', 'AT: WIP']
+            })
+      if: ${{ failure() }}
+
+    - name: REMOVE PASS Label Results for Failure
+      uses: actions/github-script@v4
+      with:
+        github-token: ${{secrets.GITHUB_TOKEN}}
+        script: |
+            github.issues.removeLabel({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              name: 'AT: CLANG-FORMAT PASS'
+            })
+      if: ${{ failure() }}
+      continue-on-error: true
+
+    - name: ADD COMMENT ABOUT FAIL
+      uses: actions/github-script@v4
+      with:
+        github-token: ${{secrets.GITHUB_TOKEN}}
+        script: |
+            github.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: '**CLANG-FORMAT TEST - FAILED** (on last commit): <br>Run > ./scripts/clang-format-test.sh using clang-format v12 to check formatting'
+            })
+      if: ${{ failure() }}
+
+#############################
+
+# Label the Results - FOR SUCCESS
+    - name: ADD PASS Label Results for Success
+      uses: actions/github-script@v4
+      with:
+        github-token: ${{secrets.GITHUB_TOKEN}}
+        script: |
+            github.issues.addLabels({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              labels: ['AT: CLANG-FORMAT PASS']
+            })
+      if: ${{ success() }}
+
+    - name: REMOVE WIP Label Results for Success
```

Comment:
So I need to learn more about actions, but is this going to run regardless of whether or not a user sets the WIP flag? Because we don't want to accidentally remove a WIP flag if it was set by the submitter.

Reply:
The AT: WIP label is one way.  Also the user can set [WIP] in the title of the PR to have the same effect.   That might be a better solution for PR's that are not wanted to be processed.